File: app/src/main/java/com/mutant/godutch/fragment/SettingsFragment.java
package com.mutant.godutch.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.google.firebase.auth.FirebaseAuth;
import com.mutant.godutch.LoginActivity;
import com.mutant.godutch.R;

/**
 * Created by evanfang102 on 2017/3/30.
 */

public class SettingsFragment extends Fragment {

    Button mButtonLogout;

    public SettingsFragment() {

    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_settings, container, false);
        mButtonLogout = (Button) view.findViewById(R.id.button_logout);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mButtonLogout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FirebaseAuth auth = FirebaseAuth.getInstance();
                auth.addAuthStateListener(new FirebaseAuth.AuthStateListener() {
                    @Override
                    public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                        if (firebaseAuth.getCurrentUser() == null) {
                            startActivity(new Intent(getActivity(), LoginActivity.class));
                            getActivity().finish();
                        }
                    }
                });
                auth.signOut();
            }
        });
    }
}

File: app/src/main/java/com/mutant/godutch/fragment/SettingsFragment.kt
package com.mutant.godutch.fragment

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.firebase.auth.FirebaseAuth
import com.mutant.godutch.LoginActivity
import com.mutant.godutch.R
import com.mutant.godutch.server.WebAgent
import com.mutant.godutch.utils.Utility.Companion.calculateExchangeRate
import kotlinx.android.synthetic.main.fragment_settings.*
import okhttp3.Call
import okhttp3.Callback
import okhttp3.Response
import java.io.IOException

/**
 * Created by evanfang102 on 2017/3/30.
 */

class SettingsFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.fragment_settings, container, false)
        return view
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        button_logout.setOnClickListener {
            val auth = FirebaseAuth.getInstance()
            auth.addAuthStateListener { firebaseAuth ->
                if (firebaseAuth.currentUser == null && activity != null) {
                    startActivity(Intent(activity, LoginActivity::class.java))
                    activity.finish()
                }
            }
            auth.signOut()
        }

        button_exchange_rate.setOnClickListener {
            WebAgent.fetchExchangeRate(object : Callback {
                override fun onResponse(call: Call, response: Response) {
                    val jsonStr = response.body().string()
                    Log.d("erJson", "MOP -> TWD:" + calculateExchangeRate(jsonStr, "USDMOP"))
                    Log.d("erJson", "HKD -> TWD:" + calculateExchangeRate(jsonStr, "USDHKD"))
                }

                override fun onFailure(call: Call, e: IOException) {
                    // TODO
                }

            })
        }
    }


}
